### ISSUE TITLE
Support for GRPC

### ISSUE THREAD

**Issue by tusharmath** [tailcallhq/tailcall#414]
Tailcall currently supports [HTTP](https://tailcall.run/docs/intro/operators/#http) as it's only data-source. Being a generic GraphQL proxy, it should be able to load data over GRPC also.

The user experience should match that of [@http](https://tailcall.run/docs/intro/operators/#http) as much as possible.

```graphql
schema @grpc(host: "grpc.service.com", port: 8080) {
  query: Query
  mutation: Mutation
}

type Query {
  user(id: ID!): User @grpc(
    service: "UserService", 
    method: "GetUser", 
    request: {user_id: "{{args.id}}"}
  )
}
```
**Technical Requirements**
- All mustache templates should work like in Http.
- Batching via data-loader should be supported.

**Additional Notes**
- Actual details about the specification would be available - #525

**Comment by ologbonowiwi** [comment]
I found [hyperium/tonic](https://github.com/hyperium/tonic); likely, we can use it.

**Comment by deepakdinesh1123** [comment]
@tusharmath Is this issue still available for work?

**Comment by tusharmath** [comment]
@deepakdinesh1123 Yes, all tickets are available until closed ;)

**Comment by thaodt** [comment]
because I'm engaging with another issue, so no time for working on this one. 
Anyone feel free to pick it, I will be here to help on guidance and code review if you want.

**Comment by ssddOnTop** [comment]
@tusharmath I think I am done with #441 (review pending) Can you assign this one to me

**Comment by A-N-uraag** [comment]
@tusharmath throwing my 🎩 in since there was no activity since 4 days. Can I get assigned?

**Comment by tusharmath** [comment]
@ssddOnTop Are you working on this?

**Comment by ssddOnTop** [comment]
> @ssddOnTop Are you working on this?

Yes

**Comment by ssddOnTop** [comment]
Still stuck on a few things, I'll make a PR in a few days

**Comment by ssddOnTop** [comment]
@tusharmath sorry I won't be able to do grpc integration

**Comment by AayushMohan** [comment]
Hey @tusharmath,  can you please assign this to me?

**Comment by tusharmath** [comment]
This issue has been attempted multiple times before and never reached completion. I would recommend only folk who have a good understanding of the project and grpc in general to attempt it. @A-N-uraag @AayushMohan let me know if you feel confident about implementing this feature. 
Alternatively working as a team with split bounty  could also work.

**Comment by Rutik7066** [comment]
Hi @tusharmath i can. And would like to work on this issue.

**Comment by amitksingh1490** [comment]
@Rutik7066 Please Go ahead assigning this to you

**Comment by Rutik7066** [comment]
Thank you @amitksingh1490

**Comment by tusharmath** [comment]
We are removing bounty from this PR. It's going to be taken up internally for now. Thank you everyone for trying 🙏

**Comment by Rutik7066** [comment]
No problem @tusharmath. I have researched and started learning package call tonic for grpc. And was looking to complete the work by this weekend.

**Comment by tusharmath** [comment]
Feel free to connect with @amitksingh1490 who has made a lot of progress in his PR.

**Comment by tusharmath** [comment]
fixed in #730
